### DISCUSSION TITLE
How to hide the grid ontop of texture images?

### DISCUSSION THREAD

**Discussion by wlwatkins** [emilk/egui#2754] [Q&A] [answered]
question in the title
![image](https://user-images.githubusercontent.com/8661949/219946490-f7e46683-a6a4-48fc-ae42-4306fe9339bc.png)

**Comment by wolro** [comment, marked as answer]
Hi, 

I am also very interested in using egui to display heatmaps based on textures, and still looking for a satisfying solution. One option is to not show the axes alltogether using the "show_axes" function when building the plot, e.g.:

```
Plot::new("heatmap demo")
    .show_axes([toggle_show_x_axes, toggle_show_y_axes])
    .show(ui, |plot_ui| {plot_ui.image(image.name("Image"))});
```

Where "toggle_show_x_axes" and "toggle_show_y_axes" are booleans. Toggling them with e.g. a checkbox, one can choose to display the axis grid or not.

An ideal solution (to me) would be some z-order parameter (as implemented in other plotting toolkits) so that one can choose the image to be above the grid. Now in the case of the egui plot functionality, this solution would have the issue that the axis tick labels are shown "inside" of the plotting window. There is an open pull request (which I am also very interested in) that would allow to display tick labels and axis labels outside, see here:

https://github.com/emilk/egui/pull/2284

Looking out for this being merged in, since the tick labels are often not very readable when displayed on top of a heatmap. And overall, egui looks fantastic for performant data visualization.

**Reply by wlwatkins** [reply]
I beleive some new updates made it work great for me.
